File: backend/Monetria.Domain/Entities/Account.cs
using Monetria.Domain.Enums;

namespace Monetria.Domain.Entities;

public class Account
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }

    public string Name { get; set; } = string.Empty;
    public AccountType Type { get; set; }
    public decimal InitialBalance { get; set; }
    public string CurrencyCode { get; set; } = "PEN";
    public bool IsActive { get; set; } = true;
    public string ColorCode { get; set; } = "#000000";

    // Visual / institution metadata
    public string? InstitutionName { get; set; }
    public string? CardLast4Digits { get; set; }
    // Credit card only
    public decimal? CreditLimit { get; set; }
    public string? CardHolderName { get; set; }
    public int? ClosingDay { get; set; }
    public int? DueDay { get; set; }

    // Wallet only
    public string? ProviderName { get; set; }
    
    public int? BillingDate { get; set; }
    public int? PaymentDay { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
}

File: apps/web/src/app/types/models/account.ts
import { AccountType } from '@monetria/enums';

export interface Account {
  id: string;
  name: string;
  type: AccountType;
  cardLast4Digits?: string; // Last 4 digits
  cardHolderName?: string;
  expiryDate?: string; // MM/YY
  institutionName?: string;
  currentBalance: number; // balance derived from transactions
  initialBalance?: number; // only used when creating an account (triggers initial transaction)
  creditLimit?: number;
  statementClosingDay?: number; // day 1-31
  paymentDueDay?: number; // day 1-31
  colorCode: string;
  currencyCode: string;
  createdAt: string;
}
